libraries/bsg_manycore_cuda.h

```diff
@@ -5,8 +5,10 @@
 
 #ifdef __cplusplus
 #include <cstdint>
+#include <ctime>
 #else
 #include <stdint.h>
+#include <time.h>
```

Comment:
how about making a stopwatch class like this?
```
#include <chrono>
#include <stdexcept>
#include "stopwatch.h"
using namespace std;
using namespace std::chrono;


/*
 *	default constructor.
 */
Stopwatch::Stopwatch()
{
	_started = false;
}

/*
 * 	start the stopwatch.
 */
void Stopwatch::start()
{
	if (_started)
	{
		throw runtime_error("Stopwatch has already started.");
	}
	_started = true;
	_start = high_resolution_clock::now();
}

/*
 * 	stop the stopwatch and return the duration.
 * 	@return amount of time elapsed in seconds.
 */
double Stopwatch::stop()
{
	if (!_started)
	{
		throw runtime_error("Stopwatch has not started.");		
	}

	duration<double> diff = high_resolution_clock::now() - _start;
	_started = false;	
	return diff.count(); 
}
```

Reply:
just an example of an interface.

Reply:
except we should make it C friendly.